Algorithms beginning

trav-d13 opened the pull request:
Addition of algorithm package starting to deal with useful algorithms. 
It is currently extended to Dijkstra's algorithm only. 
Additionally, contains test cases for the algorithm

Commit:
Missing removal statement

Commit:
Modify gitignore to ignore the gradlew file

Commit:
Addition of Dijkstra algorithm framework.
Simple graph is created to test Dijkstra's algorithm.

Commit:
Completion of test case with correct assertion
Additional hint of shortest path forming a tree structure.

Commit:
Simplification of comparison

S010MON commented on app/src/main/java/algorithms/Dijkstra.java:
```diff
@@ -0,0 +1,43 @@
+    }
+
+    /**
+    This method  calculates the shortest distance from the root node to all other nodes in the graph using Dijkstra's algorithm.
```
Consider removing "This method"  part, implied by the location

S010MON commented on app/src/test/java/algorithms/DijkstraTest.java:
```diff
@@ -0,0 +1,37 @@
+package algorithms;
+
+import org.junit.jupiter.api.Test;
+import static org.junit.jupiter.api.Assertions.*;
```
Wildcard import when only using `AssertEquals` in the file

S010MON commented on app/src/test/java/algorithms/DijkstraTest.java:
```diff
@@ -0,0 +1,37 @@
+     */
+    @Test void testSimpleGraph()
+    {
+        int[][] graph = { { 0, 4, 0, 0, 0, 0, 0, 8, 0 },
```
Sub optimal formatting for this.  try and make them inline

S010MON reviewed (approved):
Looks good, have not run it.  Some comments on formatting in review

Commit:
Formatting of Dijkstra information completed

Commit:
- Additional Dijkstra tests added
- Dijkstra example code provided and passing tests successfully.
- Dijkstra example documented and comments left at key points.
